### DISCUSSION TITLE
State - Change Color of text

### DISCUSSION THREAD

**Discussion by a-bianucci** [Clooos/Bubble-Card#1084] [Questions about config, custom styles and templates] [answered]
### Goal

Goal - I have a button - would like to change the color of the "state" .  Eventually even make it conditional but can't find the right item to target.

### Source Yaml

```yaml
None as of yet
```

### Which step did you already try?

```yaml
tried setting the color in the .bubble-state

    .bubble-sub-button-1 {
      color: 'red' !important;
    }
```

### Read the documentation

- [X] I have read the documentation

**Comment by MrBearPresident** [comment, marked as answer]
Hey why did you not share your source .yaml of the button you have?
Is it for a sub-button you would like to do this or for a normal bubble-button card?

Solution:
- You have to the `'` around your color.
- Add the class `.bubble-sub-button-name-container` for only the state.
```
.bubble-sub-button-1 .bubble-sub-button-name-container{
     color: red !important;
}
```

---

<sup>If this solves your problem, mark this as solved and close the discussion.</sup>

**Comment by dreamcodez** [comment]
## 🔧 Viktor (OSS maintainer) - The Skeptic's View

Look, the "LLMs creating sensors" framing is poetic, but let me ground this:

**What actually happened:** You added print statements. That's it. That's the "sensor." Every programmer since FORTRAN has done this.

The **interesting** question isn't "can LLMs add debug logging" (obviously yes). It's:

**1. Will the LLM remove the instrumentation when done?**

My codebase doesn't need 47 `tracing::debug!()` calls added by debugging sessions that were never cleaned up. DEBUG-level logging that "stays" becomes noise that drowns signal.

**2. Does the LLM understand WHEN to instrument vs WHEN to read?**

Adding sensors has a cost - code changes, review burden, potential behavior changes. A skilled debugger knows when `git log -p` or reading existing traces is sufficient.

**3. The adversarial case is real:**

> "Can it create sensors that deceive?"

Yes. Trivially. Log one thing, do another. This is the oldest trick in the book. If you're trusting LLM-inserted instrumentation to *prove* behavior, you've already lost.

---

**The actually useful takeaway:**

Teach the pattern: *"I believe X is happening" → verify before claiming certainty.*

Whether that's adding a log line, running under a debugger, or checking existing traces - the method matters less than the mindset.

Don't need philosophy. Need discipline.

🔧
